python/amici/petab_objective.py

```diff
@@ -1,181 +1,588 @@
+                         for key, val in condition_map_sim.items()}
+    condition_map_preeq = {key: _get_par(key, val)
+                           for key, val in condition_map_preeq.items()}
+
+    # separate fixed and variable AMICI parameters, because we may have
+    # different fixed parameters for preeq and sim condition, but we cannot
+    # have different variable parameters. without splitting,
+    # merge_preeq_and_sim_pars_condition below may fail.
+    variable_par_ids = amici_model.getParameterIds()
+    fixed_par_ids = amici_model.getFixedParameterIds()
+
+    condition_map_preeq_var, condition_map_preeq_fix = \
+        subset_dict(condition_map_preeq, variable_par_ids, fixed_par_ids)
+
+    condition_scale_map_preeq_var, condition_scale_map_preeq_fix = \
+        subset_dict(condition_scale_map_preeq, variable_par_ids, fixed_par_ids)
+
+    condition_map_sim_var, condition_map_sim_fix = \
+        subset_dict(condition_map_sim, variable_par_ids, fixed_par_ids)
+
+    condition_scale_map_sim_var, condition_scale_map_sim_fix = \
+        subset_dict(condition_scale_map_sim, variable_par_ids, fixed_par_ids)
+
+    logger.debug("Fixed parameters preequilibration: "
+                 f"{condition_map_preeq_fix}")
+    logger.debug("Fixed parameters simulation: "
+                 f"{condition_map_sim_fix}")
+    logger.debug("Variable parameters preequilibration: "
+                 f"{condition_map_preeq_var}")
+    logger.debug("Variable parameters simulation: "
+                 f"{condition_map_sim_var}")
+
+    petab.merge_preeq_and_sim_pars_condition(
+        condition_map_preeq_var, condition_map_sim_var,
+        condition_scale_map_preeq_var, condition_scale_map_sim_var,
+        condition)
+
+    logger.debug(f"Merged: {condition_map_sim_var}")
+
+    # If necessary, bring parameters to linear scale
+    if scaled_parameters:
+        # For dynamic parameters we could also change ExpData.pscale, but since
+        # we need to do it for fixed parameters anyways, we just do it for all
+        # and set pscale to linear. we can skip preequilibration parameters,
+        # because they are identical with simulation parameters, and only the
+        # latter are used from here on
+        unscale_parameters_dict(condition_map_preeq_fix,
+                                condition_scale_map_preeq_fix)
+        unscale_parameters_dict(condition_map_sim_fix,
+                                condition_scale_map_sim_fix)
+        unscale_parameters_dict(condition_map_sim_var,
+                                condition_scale_map_sim_var)
+
+    ##########################################################################
+    # variable parameters and parameter scale
+
+    # parameter list from mapping dict
+    parameters = [condition_map_sim_var[par_id]
+                  for par_id in amici_model.getParameterIds()]
+
+    edata.parameters = parameters
+
+    edata.pscale = amici.parameterScalingFromIntVector(
```

Comment:
has the problem that wrong gradients are calculated due to scaling all parameters to lin scale been resolved already? If not maybe at an issue+warning?

Reply:
Noted in docstring. Needs some more thought on how to handle that. The current implementation allows providing only a subset of parameters and using PEtab nominalValues for the rest. Not completely sure which sensitivities should be returned (only for those? or selected via Model::parameterList? or all?)

Reply:
simplest and most efficent solution would be to just return sensitivities w.r.t. requested parameters.